test/test_rake_task_arguments.rb

```diff
@@ -54,6 +54,11 @@ def test_to_hash
     assert_equal 0,  h.fetch(:one)
   end
 
+  def test_deconstruct_keys
```

Comment:
You should omit older versions.
I'm not sure if ruby 3.0 should be supported, because one-line pattern matching was experimental.

```suggestion
  def test_deconstruct_keys
    omit "Pattern matching is not supported yet" if RUBY_VERSION < "3.1"
```

Reply:
Agreed on both counts with thanks, updated and rebased